### DISCUSSION TITLE
Declare your Ad ID permission

### DISCUSSION THREAD

**Discussion by marioshtika** [admob-plus/admob-plus#508] [General]
Yesterday I got an email from the Google Play team about declaring the Ad ID permission.

> Hello Google Play Developer,
> 
> Last July, we announced **Advertising policy** changes to help bolster security and privacy. We added new restrictions on identifiers used by apps that target children.
> 
> When users choose to delete their advertising ID in order to opt out of personalization advertising, developers will receive a string of zeros instead of the identifier if they attempt to access the identifier. This behavior will extend to phones, tablets, and Android TV starting April 1, 2022.
> 
> We also announced that you need to declare an AD_ID permission when you update your app targeting API level to 31 (Android 12). Today, we are sharing that we will give developers more time to ease the transition. We will require this permission declaration when your apps are able to target Android 13 instead of starting with Android 12.
> 
> **Action Items**
> - If you use an advertising ID, you must declare the **AD_ID Permission** when your app targets Android 13 or above. Apps that don’t declare the permission will get a string of zeros. Note: You’ll be able to target Android 13 later this year.
> - If your app uses an SDK that has declared the Ad ID permission, it will acquire the permission declaration through manifest merge.
> - If your app’s **target audience** includes children, you must not transmit Android Advertising ID (AAID) from children or users of unknown age.

Is this something new?
Is this already implemented?

Thank you in advance.

**Comment by r2uan** [comment]
Just received a mail from google play saying about this too. Can anyone confirm if this does use the advertising id for google as well as apple?

**Reply by merbin2012** [reply]
@r2uan This plugin currently not yet support advertisement id requirement. This feature only available from admob 20.5.0, but this plugin supports admob 20.4.0. 

Also, this give error if you are having latest Cordova-android (ver 11.0).

**Reply by Carlosps** [reply]
> @r2uan This plugin currently not yet support advertisement id requirement. This feature only available from admob 20.5.0, but this plugin supports admob 20.4.0.
> 
> Also, this give error if you are having latest Cordova-android (ver 11.0).

Thank you for letting us know about it.

What if we add the ```<uses-permission android:name="com.google.android.gms.permission.AD_ID" />``` in our Android manifest?
Wouldn't that help our app to be in accordance with that warning?
